### ISSUE TITLE
fotka kola

### ISSUE THREAD

**Issue by banterCZ** [Clevis/ReKola-Public#5]
Co takhle přidat k detailu kola i jeho fotku?

**Comment by VitekJezek** [comment]
nad tím jsme dokonce sami uvažovali. Ono my o těch kolech v DB máme spousta infa; takže první krok je to nějak dobře vypsat, druhý krok nad tím filtrovat a v třetím kroku bych viděl přidání fotky, bude-li nás to stále pálit ; )

**Comment by hnykda** [comment]
Ahoj.
Chtěl bych k tomuto feature request připsat, jak bych fotografie v app ocenil já. Berte to prosím jako vlídný a co nejkonstruktivnější návrh uživatele.

## Problém

Textový popis kol je v některých případech nedostatečný. Měřím 193+ cm a tudíž uvítám spíše větší kolo a s vyšší sedačkou i za cenu pohodlnosti (např. silničky oproti dámskému). V textovém popisu toto obsáhnuto není a ani to není tak úplně možné.

## Navrhované řešení

Možnost přidat k danému kolu fotku a to na několika úrovních:
- _Hlavní fotografie_ : Tato fotka by byla hlavní fotografie kola - v ucházející kvalitě za dobrého osvícení vložena adminy/repairery (ty by měli možnost jí aktualizovat, kdyby po nějaké opravě vypadalo kolo jinak).
- _Fotografie vložená uživatelem_ : Poslední "půjčující"  má možnost přidat fotografii kola, ze dvou důvodů - aktuální stav kola, který může sloužit jak repairers v případě nahlášení problému, nebo dalším zájemcům o půjčení kola (ví jak aktuálně vypadá - např. se snížila sedačka, je špinavé...). V poslední řadě se dá vyfotit z větší dálky k upřesnění místa, kde kolo stojí. 

## Poznámky
- K fotkám by mělo být možné přidávat komentáře. Základní nabídka přednastavených komentářů by též měla být k dispozici. Mimo jiné by obsahovala "Upřesnění pozice kola", "Problém na kole", "Detail kola"...
- Další zájemce o kolo by mohl mít možnost vidět fotografie od posledního půjčujícího a pokud by poslední půjčující žádnou fotografii neudělal, mělo by to být viditelné (místo fotografie text: "Poslední půjčující neudělal žádnou fotografii kola"). 
- U každého kola by měla být galerie předchozích fotek.

**Comment by Piskvor** [comment]
Mně se to líbí; ale teda připadá mi to jako hodně masitá nadstavba nad aplikací (jak z hlediska datových přenosů, tak z hlediska vývojové náročnosti).

**Comment by hnykda** [comment]
Nepochybně. Navíc nesnáším, když si nějaká aplikace bez dovolení načítá několik 3 mb fotek. Proto by bylo nutné to udělat rozumně - např. aby se v základu zobrazovala jen hlavní fotografie kola (100x100px je tak 10kB ) s možností rozkliknout detail na řekněme 1024x720 (200-400 kb při snížení kvality, jpg) . "Poslední" fotografie, vzhledem k rozlišení dnešních telefonů, též stačí v nějakém podobném formátu (miniatura + větší). 
Galerie by nemusela být nekonečná a mohlo by zůstávat jen pár posledních fotografií. A zmenšovat i 6 mb fotografie zvládne i RPi (vlastní zkušenost, např. epeg).
Vývojovou náročnost nejsem schopen odhadnout. Nejblíže aplikacím jsem dělal webovou app v Djangu, kde výše uvedená věc funguje a skládá se z jednoho pluginu + 7 řádků kodu v Pythonu.

**Comment by Piskvor** [comment]
No, ona uz se v zakladu nacita mapa+jeji dependence, to je taky docela bumbrlicek. Spis bych to videl na nejaky rozkliknutelny detail, nez ze by se to zobrazovalo defaultne. (co do narocnosti vyvoje si to taky netroufam odhadovat, ale "sedm radku a plugin" mi tu pripada velmi, velmi optimisticke)

**Comment by VitekJezek** [comment]
díky za poznámky. Co říkáte na tohle kompromisní řešení?
- po vrácení kola (https://cloudup.com/cq0593jVTvt - pracovní verze wireframe; ty body jsou nějaká "karma" jak tomu pracovně říkáme) a odkliku "sloužilo dobře => vše OK" by se v některých případech (nebo možá všech? chce to měřit) objevil nepovinný dotazník - třeba odklikávání pomocí hvězdiček. Musí to být ultrarychlý. V něm by byli tyto otázky:
- celkový stav kola: špica … měl by se na to mrknout servisman
- pro jak velké lidi je? mini … maxi
- jak jezdí do kopce? spíš po rovině … terení koza
- chceš k němu něco říct? Jaké se ti stalo story? Kam tě rekolo dovezlo? Co jdeš dělat? Poděl se! 
  - a tohle by byl komentář. Ten by šel přidat i normálně ke kolu z jeho detailu. A u komentáře by šlo přidat fotku. 
  - nějaký galerie na detailu (chce dopromyslet); možnost nějakou fotku z galerie "pinnout" jako hlavní.

Na samotnou screenu s vracením to nechci úplně dávat, protože tam chceme, aby to opravdu každý vrátil ASAP. Je to ten hlavní workflow. Když pak třeba mám čas, tak můžu udělat něco víc (po cestě do kavárny / po schodech do práce). Třeba i zpracovat a nahrát fotku. Nebo story. Ale zdržovat tím před vrácením kola je proti systému rychlého vrácení do oběhu.

Lajk or hejt?

**Comment by hnykda** [comment]
+1

**Comment by Piskvor** [comment]
Jo, vypadá to dobře; jen bych na tý obrazovce ještě zdůraznil "tohle už je nepovinná omáčka kolem" (pun not intended).
